src/Controls/StreetControls.js

```diff
@@ -102,19 +239,123 @@ class StreetControls extends FirstPersonControls {
     }
 
     /**
-     * Move the camera
+     * Method called when user click on the ground.</br>
+     * Note that this funtion contains default values that can be overrided, by overriding this function.
+     *
+     * @param {THREE.Vector3} position - The position
+     */
+    onClickOnGround(position) {
+        // compute a new position to go : 4 meter up from the picked position on the ground
+        const up = position.clone().normalize();
+        position.add(up.multiplyScalar(4));
+
+        // compute time to go there
+        const distance = this.camera.position.distanceTo(position);
+        // 500 millis constant, plus an amount of time depending of the distance (but not linearly)
+        const time = 500 + Math.sqrt(distance) * 300;
+
+        // move the camera
+        this.moveCameraTo(position, time);
+    }
+
+    /**
+     * Method called when user click on oject that is not the ground.</br>
+     * Note that this function contains default values that can be overrided, by overriding this function.
+     *
+     * @param {THREE.Vector3} position - The position
+     */
+    onClickOnWall(position) {
+        const distance = this.camera.position.distanceTo(position);
+
+        // can't click on a wall that is at 1km distance.
+        if (distance < 1000) {
```

Comment:
This value is very arbitrary, could you set it in a variable in the constructor ?

Reply:
ok